languages/python/docs/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,49 @@
-# How to implement an Python concept exercise
+# How to implement a Python concept exercise
 
-TODO: describe how to implement a concept exercise for the Python track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This document describes the steps required to implement a concept exercise for the Python track. As this document is generic, the following placeholders are used:
 
-[csharp-implementing]: ../../csharp/docs/implementing-a-concept-exercise.md
+- `<SLUG>`: the name of the exercise in kebab-case (e.g. `anonymous-methods`).
+- `<NAME>`: the name of the exercise in PascalCase (e.g. `AnonymousMethods`).
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue.
+
+To implement a concept exercise, the following files need to be created:
+
+```
+languages
+└── Python
+    └── exercises
+        └── concept
+            └── <SLUG>
```

Comment:
This structure seems to have been inspired by that of the Haskell track. I think it would make sense to use the same structure here as the Python track is already using (e.g. https://github.com/exercism/python/tree/master/exercises/allergies). Unless you've all agreed to change it of course.

Reply:
@ErikSchierboom yes it was inspired by that one. I left it the way it was because I thought it was the new one proposed for v3. I'll change it as soon as I'm sure how it will be. Thx!

Reply:
I'm going to propose something more like:

```
languages
└── python
    └── exercises
        └── concept
            └── <SLUG>
                ├── .docs
                |   ├── instructions.md
                |   ├── introduction.md
                |   ├── hints.md
                |   └── after.md
                ├── .meta
                |   ├── config.json
                |   └── example.py
                ├── <NAME>.py
                └── <NAME>_test.py
        └── practice
            └── <SLUG>
                ├── .docs
                |   ├── instructions.md
                |   ├── introduction.md
                |   ├── hints.md
                |   └── after.md
                ├── .meta
                |   ├── config.json
                |   └── example.py
                ├── <NAME>.py
                └── <NAME>_test.py
```

This mimics what the C# track has chosen, but remains consistent with what all our tooling is currently set up to expect (they all expect a path as far as `<SLUG>`). Only change from what we had before is the division into `concept` and `practice` exercises.

@cmccandless ?

Reply:
> I left it the way it was because I thought it was the new one proposed for v3.

It was a good intuition! Tracks are free though to set things up as they like, except for the non-track specific files like `.meta/config.js` and the files in the `.docs` directory.

Reply:
I agree with what @yawpitch has proposed. This would minimize required changes to tooling for the time being.

Reply:
Great, thanks @cmccandless.

@DavidGerva can you please update your PR to match the above?

Reply:
Yes I'm doing it!